### DISCUSSION TITLE
Create a "works with Authelia" matrix to dramatically increase uptake of this project

### DISCUSSION THREAD

**Discussion by timdonovanuk** [authelia/authelia#5201] [Ideas]
Just my $0.02!

To massively increase people using Authelia, a list of commonly self hosted apps that are confirmed to work with Authelia would drastically increase visbility and usage (and those that only partially work, have instructions or notes for each).

I suggest starting with the linuxserver list of applications: https://fleet.linuxserver.io/
 plus some of the crowd favorites such as Proxmox, Portainer, Grafana etc...

If people can see that their self hosted apps are supported and confirmed working, they are much more likely to give Authelia a go. Right now, its unknown, and people are not going to try Authelia unless they know the majority of their apps will work.

**Comment by clems4ever** [comment]
Hello @timdonovanuk ,

thanks for the suggestion. Are you in this situation right now to decide whether to test Authelia? What apps do you plan to protect?

I agree that what you suggest could convince (massively maybe not) some users to test Authelia. However, it's not as easy as putting some ticks in a matrix. This matrix should be a shared contributions because every environment is different, and the core team does not have the resources to test them all. Moreover, the state for one app is guaranteed to be true only at a certain point in time. Trying to keep this matrix up-to-date and documenting with instructions as you suggest does not come for free and we are not staffed to do better than best effort on this unfortunately.
Another approach could be to create suites for each app we want to make sure we support and run them regularly but again, doing that would take both lot of time to put in place and a lot of resources to run regularly and it's not really bringing a lot of value to our existing community.

So maybe an idea could be to add a dedicated page for this matrix with the first items under https://www.authelia.com/docs/community/ but we need to mention that this matrix is a community work and the core team cannot guarantee the full relevance of the information even though we'll do our best to maintain it.

**Comment by trallnag** [comment]
In what way can an app "support" Authelia?

**Comment by clems4ever** [comment]
That's a very good question @trallnag .

Theoretically, Authelia protects any HTTP API whatever the application exposing it. Authelia does not know anything about the app it is protecting because ACLs are based on domains and paths. (Note that the previous statement is not true for OpenID Connect applications for which the administrator must define the clients in the configuration). So, beside OIDC, there should not be any concept of supported app from an authentication point of view.
However we detected some technical issues in the past with some applications (CORS preflight not handled properly, websockets, bug with Home Assistant) but those are technical bugs that we definitely analyze and fix and until we fix them they are "not supported" but I would say this is an edge case and so I might not consider maintaining a list of apps not working because of bugs but rather open dedicated issues and fix the bugs instead.

Now, regarding authorization models, things are a bit trickier. Authelia performs authentication and must sometimes forward authentication information to the app it protects for the app, based on the forwarded identity of the authenticated user, to resolve the authorizations (this avoid double authentication to get the permissions). This is usually done transmitting the X-Forwarded-User and X-Forwarded-Groups headers containing the information but not all apps support this way of trusting the proxy and it's also not secure because any client can act on behalf of someone else from within the internal network if ACLs are not set up to fix that. That's why we decided to move to OpenID Connect, mainly because the backend can either consumes the scopes set in the OAuth2 transaction or consume the ID Token to resolve the permissions in a secure way. So maybe the supported matrix could be based on this kind of information (support for X-Forwarded-User) but supporting this is not a pre-requesite for all apps so it might not be the best idea to list that either.

@timdonovanuk, did you have that in mind when opening this issue?

**Comment by MarcHagen** [comment]
@clems4ever Netbox is also compatible with Authelia, is it possible to add this too ?

**Comment by james-d-elliott** [comment]
I think this is solved by us adding the integrations area. If users would like to add specifically tested setups then we welcome community contributions.

https://www.authelia.com/integration/prologue/introduction/

**Reply by MarcHagen** [reply]
This is really awesome! It helps all of the community in this way. I'll see if i can create a PR for Netbox aswell to have that added there.
